src/test/ui/impl-trait/impl-fn-predefined-lifetimes.rs

```diff
@@ -0,0 +1,13 @@
+#![feature(impl_trait_in_fn_trait_return)]
+use std::fmt::Debug;
+
+fn a<'a>() -> impl Fn(&'a u8) -> (impl Debug + '_) {
```

Comment:
Hmm, I'm kind of surprised by this error -- I expect this function to be equivalent to the one below...?

Reply:
To be honest: I do not know what's going on 😄 

Note though that the same function written with a normal trait also [produces an error](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=95f0d8c2ab6fd8d27330e30deee08c72):
```rust
trait Trait<T> {
    type Item;
}

impl<T> Trait<T> for () {
    type Item = ();
}

fn a<'a>() -> impl Trait<&'a u8, Item = impl Debug + '_> {}
// ^^^ doesn't compile, for better or worse
```

So I don't think it's an issue related to this PR.

Reply:
This test makes sense to me, at least with the current way we're implementing lifetime elision. 

The `'_` lifetime here is looking at the arguments of the function (i.e. an empty list) -- not at the arguments of the `impl Fn(..)` -- for input-position lifetimes to un-elide `'_` with. Since there are none, we error.

I'm not sure if this is the *desired* behavior, but I think this doesn't block landing this behind a feature gate.

Reply:
Also, the fact that this fails to compile currently (cannot resolve opaque type) is due to https://github.com/rust-lang/rust/issues/101968#issuecomment-1252815586

Reply:
Also this is broken by the same regression as https://github.com/rust-lang/rust/issues/101968#issuecomment-1252815586